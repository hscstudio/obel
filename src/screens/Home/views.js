import objstr from 'obj-str'
import PropTypes from 'prop-types'
import React from 'react'
import logo from './logo.svg'

export const Button = ({ children, on, onToggle }) => (
  <button
    className={
      'hover:underline ' +
      objstr({
        'text-blue-400': !on,
        'text-green-400': on
      })
    }
    onClick={() => onToggle()}
  >
    {children}
  </button>
)

Button.propTypes = {
  children: PropTypes.node.isRequired,
  on: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired
}

export const Logo = () => <img alt='logo' className='spin h-64' src={logo} />
