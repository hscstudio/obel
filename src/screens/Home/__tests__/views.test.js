import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import { Button } from '../views'

describe('Button', () => {
  it('renders differently based on the toggle value ', () => {
    expect(<Button />).toMatchDiffSnapshot(<Button on />)
  })

  it('calls onToggle correctly on click', () => {
    const onToggle = jest.fn()
    const { getByText } = render(
      <Button onToggle={onToggle}>Learn React</Button>,
    )

    fireEvent.click(getByText(/learn react/i))
    expect(onToggle).toHaveBeenCalledTimes(1)
    expect(onToggle).toHaveBeenCalledWith()
  })
})
