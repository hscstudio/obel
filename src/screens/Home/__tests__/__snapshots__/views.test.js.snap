// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders differently based on the toggle value  1`] = `
Snapshot Diff:
- <Button />
+ <Button on={true} />

  <button
-   className="hover:underline text-blue-400"
+   className="hover:underline text-green-400"
    onClick={[Function onClick]}
  />
`;
